akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/HandlerAdapter.scala

```diff
@@ -11,13 +11,23 @@ import scala.jdk.CollectionConverters._
 
 import akka.Done
 import akka.annotation.InternalApi
+import akka.projection.internal.ActorHandlerInit
 import akka.projection.javadsl
 import akka.projection.scaladsl
 
+object HandlerAdapter {
+  def apply[Envelope](delegate: javadsl.Handler[Envelope]): scaladsl.Handler[Envelope] = {
+    delegate match {
+      case a: javadsl.ActorHandler[Envelope, Any] @unchecked => new ActorHandlerAdapter[Envelope](a)
+      case _                                                 => new HandlerAdapter(delegate)
+    }
+  }
+}
```

Comment:
This adapter should be in core module instead. I think I moved it in one of the open PRs I have. Anyway, I can fix it when I resume my work on that other PR.